### ISSUE TITLE
Pressing [e or ]e closes current fold.

### ISSUE THREAD

**Issue by MeanEYE** [tpope/vim-unimpaired#96]
Like the title says. If you press `[e` or `]e` current fold will close and line will be moved. Needless to say this is hindering as every time you have to open fold after moving the line effectively negating benefits of easy line moving.

**Comment by tpope** [comment]
Can't reproduce.

**Comment by MeanEYE** [comment]
Hm, let me see if this is due to `foldmethod`.

**Comment by MeanEYE** [comment]
I can only reproduce when `foldmethod=indent`.

**Comment by BohrShaw** [comment]
I encountered the same issue from time to time especially the session lasts very long. But I also can't reproduce it now. The `foldmethod` could be irrelevant as I have it on various types.

**Comment by BohrShaw** [comment]
This is actually caused by e.g. a simple `move +` command when some specific yet unknown settings present, which causes a fold recomputing or a redraw I guess.

I tried with the command `execute 'set fdm=manual|move+|set fdm='.&fdm` and it works properly. But I doubt if this would be a solution. I probably should report to "vim_dev" but it's not deliberately reproducible yet.

**Comment by poppyschmo** [comment]
>This is actually caused by e.g. a simple `move +` command

Agree. I've run into this only recently when `'foldmethod'` is set to anything other than  `manual`. Only seems to be triggered by `:move` (normal and visual modes) and not similar line-wise commands like `:t`, for instance. As mentioned, this is most certainly unrelated to unimpaired.

**Comment by jebaum** [comment]
relevant issue: https://github.com/vim/vim/issues/687

it is a bug in vim. there is a workaround in that issue thread which unimpaired could add if @tpope was so inclined. I could understand not wanting to for obvious reasons, but on the other hand, who knows how long it will be before this is fixed?

**Comment by tpope** [comment]
I must have upgraded to a Vim version that's broken because this has been biting me hard lately. I am willing to entertain a workaround of moderate grossness.

**Comment by jebaum** [comment]
Working on a PR now

**Comment by francozappa** [comment]
Hi @jebaum, any updates about the PR?

**Comment by IanEdington** [comment]
I'm getting this too. The issue has come up on stack overflows vi and vim community: http://vi.stackexchange.com/questions/3174/problem-moving-lines-vertically-closes-folds

**Comment by kurkale6ka** [comment]
I am having this problem too:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Nov 15 2016 18:04:06)
Included patches: 1-86
```
or
```
NVIM v0.2.0-280-ge21aef1e1
```

**Comment by scanny** [comment]
This bit me with a new update to MacVim and my plugins. Reminds me why I put off upgrades as long as possible :)

Anyway, a fix inspired by the SO answer @IanEdington linked to worked for me:

```vim
nmap - zi[ezi
nmap _ zi]ezi
```

Thanks Ian! :)

**Comment by adriaanzon** [comment]
Or you can use @jebaum's [vim-move-bugfix](https://github.com/jebaum/vim-unimpaired/tree/vim-move-bugfix) branch with the fix from #110
